src/Provider/AbstractProvider.php

```diff
@@ -355,8 +355,13 @@ protected function getAuthorizationParameters(array $options)
         // Store the state as it may need to be accessed later on.
         $this->state = $options['state'];
 
+        // Business code layer might set a different redirect_uri parameter
+        // depending on the context, leave it as-is
+        if (!isset($options['redirect_uri'])) {
```

Comment:
Should this be `!isset` or `empty`? Is an empty, but defined, `redirect_uri` valid?